repose-aggregator/components/filters/keystone-v2-filter/src/main/scala/org/openrepose/filters/keystonev2/KeystoneV2Authorization.scala

```diff
@@ -32,18 +32,15 @@ import scala.util.{Failure, Success, Try}
 
 object KeystoneV2Authorization extends LazyLogging {
 
-  def doAuthorization(config: KeystoneV2Config, request: HttpServletRequestWrapper, validToken: ValidToken, endpoints: => Try[EndpointsData]): AuthorizationInfo = {
-    lazy val tenantToMatch: String =
-      Option(config.getTenantHandling.getValidateTenant).flatMap(validateTenantConfig =>
-        Option(validateTenantConfig.getUriExtractionRegex).flatMap(uriExtractionRegexList =>
-          uriExtractionRegexList.asScala.toStream.map(_.r).flatMap(uriExtractionRegex =>
-            request.getRequestURI match {
-              case uriExtractionRegex(tenantId, _*) => Option(tenantId)
-              case _ => None
-            }
-          ).headOption
-        )).getOrElse(throw UnparseableTenantException("Could not parse tenant from the URI"))
+  val handleFailures: PartialFunction[Try[Unit.type], KeystoneV2Result] = {
+    case Failure(e: InvalidTenantException) => Reject(SC_UNAUTHORIZED, Some(e.getMessage))
+    case Failure(e: UnauthorizedEndpointException) => Reject(SC_FORBIDDEN, Some(e.getMessage))
+    case Failure(e: UnparseableTenantException) => Reject(SC_UNAUTHORIZED, Some(e.getMessage))
+  }
 
+  // TODO: Take tenants to roles mapping instead of token to calculate scoped roles.
```

Comment:
Doing this is kind of necessary to drop the requirement of a token request attribute for the authorization filter (without refactoring this a bunch, but since we're doing the map anyway, why not use it). Turns out, though, that we only need the tenant-to-role mapping and endpoints to do all of our authorization stuff.